hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/BasicFormatterImpl.java

```diff
@@ -390,6 +196,268 @@ private void newline() {
 			}
 			beginLine = true;
 		}
+
+		private enum ProcessEnum {
+
+			PROCESS_COMMA_AFTER_BY_FROM_SELECT {
+				@Override
+				public void process(FormatProcess p) {
+					if ( p.afterByOrSetOrFromOrSelect && ",".equals( p.token ) ) {
+						p.out();
+						p.newline();
+						return;
+					}
+					PROCESS_COMMA_AFTER_ON.process( p );
+				}
+			},
+
+			PROCESS_COMMA_AFTER_ON {
+				@Override
+				public void process(FormatProcess p) {
+					if ( p.afterOn && ",".equals( p.token ) ) {
+						p.out();
+						p.indent--;
+						p.newline();
+						p.afterOn = false;
+						p.afterByOrSetOrFromOrSelect = true;
+						return;
+					}
+					PROCESS_OPEN_PAREN.process( p );
+				}
+			},
+
+			PROCESS_OPEN_PAREN {
+				@Override
+				public void process(FormatProcess p) {
+					if ( "(".equals( p.token ) ) {
+						if ( isFunctionName( p.lastToken ) || p.inFunction > 0 ) {
```

Comment:
Super minor. It might be a good practice to swap the operators around `||` to boost performance in case of `p.inFunction > 0` for it seems `isFunctionName( p.lastToken )` is more expensive ones than `p.inFunction > 0`:
```
if ( p.inFunction > 0 || isFunctionName( p.lastToken ) ) {
    ... ...
}
```
The rationale is when `p.inFunction > 0`, the more expensive `isFunctionName( p.lastToken )` would be skipped due to short circuiting.

Feel free to ignore this comment though.

Reply:
IMHO, unless we can predict that `p.inFunction > 0` being true happens more often (in fact the sqls to be formatted can be very random, plus, `isFunctionName( p.lastToken )` being true is the reason of  `p.inFunction > 0` being true), otherwise  the only thing I can do is hoping that the JIT compiler can do something about it :)